src/main/java/olivero/tasks/Deadline.java

```diff
@@ -0,0 +1,45 @@
+package olivero.tasks;
+
+import java.time.LocalDateTime;
+
+import olivero.parsers.DateParser;
+
+
+/**
+ * Represents a Deadline task.
+ */
+public class Deadline extends Task {
+    private final LocalDateTime endDate;
+
+    /**
+     * Constructs a {@code Deadline} object with the given task description, end date
+     * and initial completion status.
+     *
+     * @param description The description of the deadline task.
+     * @param endDate The deadline/end date of the deadline task.
+     * @param isDone The initial completion status of the deadline task.
+     */
+    public Deadline(String description, LocalDateTime endDate, boolean isDone) {
+        super(description, isDone);
+        this.endDate = endDate;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toFormattedString() {
+        int doneStatus = isDone() ? 1 : 0;
+        String dateString = DateParser.asInputDateString(endDate);
+        return "D" + " | " + doneStatus + " | " + getDescription() + " | " + dateString;
```

Comment:
```suggestion
        return "D" + " | " + super() + " | " + dateString;
```

I would recommend extracting `doneStatus + " | " + getDescription()` in the subclasses to the Task superclass and calling super() instead to maintain the abstraction barrier (more so for the isDone attribute since it does not seem to be accessed anywhere else)